frontend/src/metabase/visualizations/echarts/cartesian/model/series.ts

```diff
@@ -333,15 +333,19 @@ export const getStackedLabelsFormatters = (
   dataset: ChartDataset,
   settings: ComputedVisualizationSettings,
   renderingContext: RenderingContext,
-): StackedSeriesFormatters => {
+): {
+  formatters: StackedSeriesFormatters;
+  compactStackedSeriesByDataKey: DataKey[];
+} => {
   const formatters: StackedSeriesFormatters = {};
+  const compactStackedSeriesByDataKey: DataKey[] = [];
```

Comment:
One more place with dictionary like naming.
Also, should it be stack name "area" | "bar" here instead of the first series?

Reply:
It works either way. 

I can perform the "should we compact the y-axis" check by:
1. which axis the `stackModel` belongs to
2. by taking a `dataKey` that is a part of that stack (e.g. the first data key in that stacked series) and checking which axis it belongs to.

Since `getSeriesLabelsFormatters` is going to the use the 2nd method of checking by `dataKey` (no other option is available), I opted to use `dataKey`s with `getStackedLabelsFormatters`, so that I do not have to keep track of two different paradigms. I can merge them into one array since they are the same concept and let that one check to do all the work.

---
I do not believe there are any performance concerns, so I plan to keep it that way unless you see a different concern.